service/history/api/describeworkflow/api.go

```diff
@@ -368,6 +382,201 @@ func buildCallbackInfo(
+				tag.Error(err),
+			)
+			return nil, serviceerror.NewInternal("failed to construct describe response")
+		}
+		if callbackInfo == nil {
+			continue
+		}
+		result = append(result, callbackInfo)
+	}
+
+	return result, nil
+}
+
+// buildCallbackInfosFromChasm reads callbacks from the CHASM tree and converts them to API format.
+func buildCallbackInfosFromChasm(
+	ctx context.Context,
+	namespaceID namespace.ID,
+	mutableState historyi.MutableState,
+	executionInfo *persistencespb.WorkflowExecutionInfo,
+	executionState *persistencespb.WorkflowExecutionState,
+	outboundQueueCBPool *circuitbreakerpool.OutboundQueueCircuitBreakerPool,
+	logger log.Logger,
+) ([]*workflowpb.CallbackInfo, error) {
+	chasmTree := mutableState.ChasmTree()
+	chasmCtx := chasm.NewContext(ctx, chasmTree.(*chasm.Node))
+
+	rootComponent, err := chasmTree.ComponentByPath(chasmCtx, nil)
+	if err != nil {
+		logger.Error(
+			"failed to get workflow component from CHASM tree",
+			tag.WorkflowNamespaceID(namespaceID.String()),
+			tag.WorkflowID(executionInfo.WorkflowId),
+			tag.WorkflowRunID(executionState.RunId),
+			tag.Error(err),
+		)
+		return nil, serviceerror.NewInternal("failed to construct describe response")
+	}
+
+	wf, ok := rootComponent.(*chasmworkflow.Workflow)
+	if !ok {
+		logger.Error(
+			"expected workflow component but got different type",
+			tag.WorkflowNamespaceID(namespaceID.String()),
+			tag.WorkflowID(executionInfo.WorkflowId),
+			tag.WorkflowRunID(executionState.RunId),
+		)
+		return nil, serviceerror.NewInternal("failed to construct describe response")
+	}
+
+	result := make([]*workflowpb.CallbackInfo, 0, len(wf.Callbacks))
+	for _, field := range wf.Callbacks {
+		callback, err := field.Get(chasmCtx)
+		if err != nil {
+			logger.Error(
+				"failed to load callback from CHASM tree",
+				tag.WorkflowNamespaceID(namespaceID.String()),
+				tag.WorkflowID(executionInfo.WorkflowId),
+				tag.WorkflowRunID(executionState.RunId),
+				tag.Error(err),
+			)
+			return nil, serviceerror.NewInternal("failed to construct describe response")
+		}
+
+		callbackInfo, err := buildCallbackInfoFromChasm(namespaceID, callback, outboundQueueCBPool)
+		if err != nil {
+			logger.Error(
+				"failed to build callback info from CHASM callback",
+				tag.WorkflowNamespaceID(namespaceID.String()),
+				tag.WorkflowID(executionInfo.WorkflowId),
+				tag.WorkflowRunID(executionState.RunId),
+				tag.Error(err),
+			)
+			return nil, serviceerror.NewInternal("failed to construct describe response")
+		}
+		if callbackInfo == nil {
+			continue
+		}
+		result = append(result, callbackInfo)
+	}
+
+	return result, nil
+}
+
+// buildCallbackInfoFromChasm converts a single CHASM callback to API format.
+func buildCallbackInfoFromChasm(
+	namespaceID namespace.ID,
+	callback *chasmcallback.Callback,
+	outboundQueueCBPool *circuitbreakerpool.OutboundQueueCircuitBreakerPool,
+) (*workflowpb.CallbackInfo, error) {
+	if callback.Callback.GetNexus() == nil {
```

Comment:
nit: prefer using getters when working with protos


```suggestion
	if callback.GetCallback().GetNexus() == nil {
```